modules/richAudienceBidAdapter.js

```diff
@@ -0,0 +1,138 @@
+    return validBidRequests.map(bid => {
+      var payload = {
+        bidfloor: bid.params.bidfloor,
+        ifa: bid.params.ifa,
+        pid: bid.params.pid,
+        supplyType: bid.params.supplyType,
+        currencyCode: config.getConfig('currency.adServerCurrency'),
+        auctionId: bid.auctionId,
+        bidId: bid.bidId,
+        BidRequestsCount: bid.bidRequestsCount,
+        bidder: bid.bidder,
+        bidderRequestId: bid.bidderRequestId,
+        tagId: bid.adUnitCode,
+        sizes: bid.sizes.map(size => ({
+          w: size[0],
+          h: size[1],
+        })),
+        referer: (typeof bidderRequest.refererInfo.referer != 'undefined' ? encodeURIComponent(bidderRequest.refererInfo.referer) : null),
+        numIframes: (typeof bidderRequest.refererInfo.numIframes != 'undefined' ? bidderRequest.refererInfo.numIframes : null),
+        transactionId: bid.transactionId,
+        timeout: config.getConfig('bidderTimeout'),
+      };
+
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        payload.gdpr_consent = bidderRequest.gdprConsent.consentString;
+        payload.gdpr = bidderRequest.gdprConsent.gdprApplies; // we're handling the undefined case server side
+      } else {
+        payload.gdpr_consent = '';
+        payload.gdpr = null;
+      }
+
+      var payloadString = JSON.stringify(payload);
+
+      var endpoint = 'https://shb.richaudience.com/hb/';
+
+      return {
+        method: 'POST',
+        url: endpoint,
+        data: payloadString,
+      };
+    });
+  },
+  /***
+     * Read the response from the server and build a list of bids
+     * @param {serverResponse} Response from the server.
+     * @param {bidRequest} Bid request object
+     * @returns {bidResponses} Array of bids which were nested inside the server
+     */
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = [];
+
+    var response = serverResponse.body;
+
+    try {
+      if (response) {
+        var bidResponse = {
+          requestId: JSON.parse(bidRequest.data).bidId,
+          cpm: response.cpm,
+          width: response.width,
+          height: response.height,
+          creativeId: response.creative_id,
+          mediaType: response.media_type,
+          netRevenue: response.netRevenue,
+          currency: response.currency,
+          ttl: response.ttl,
+        };
+
+        if (response.media_type === 'video') {
+          bidResponse.vastXml = response.vastXML;
+        } else {
+          bidResponse.ad = response.adm
+        }
+
+        bidResponses.push(bidResponse);
+      }
+    } catch (error) {
+      utils.logError('Error while parsing smart server response', error);
+    }
+    return bidResponses
+  },
+  /***
+     * User Syncs
+     *
+     * @param {syncOptions} Publisher prebid configuration
+     * @param {serverResponses} Response from the server
+     * @returns {Array}
+     */
+  getUserSyncs: function (syncOptions, serverResponses) {
```

Comment:
You can pass in a `gdprConsent` param which you can use to send consent data if needed.

I see you use the `serverResponses` object to achieve the same, but in the else case you could utilize this `gdprConsent` data if wanted.

I am unsure how RichAudience wants to handle this but though I would let you know.

Example of adapter using this: https://github.com/prebid/Prebid.js/blob/master/modules/rubiconBidAdapter.js